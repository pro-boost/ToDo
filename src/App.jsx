import "./App.css";
import { useState } from "react";
import InputTasks from "./Components/InputTasks";
import DisplaytTasks from "./Components/displayTasks";
import CountTask from "./Components/CountTask";
function App() {
  const [task, setTask] = useState("");
  const [tasks, setTasks] = useState([]);
  const [count, setCount] = useState(0);

  const handleInputChange = (e) => {
    setTask(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (task.trim() === "") return;
    setTasks([...tasks, task]);
    setTask("");
    setCount(count + 1);
  };
  return (
    <>
      <h1>To Do List :</h1>
      <InputTasks
        task={task}
        handleInputChange={handleInputChange}
        setTask={setTask}
        handleSubmit={handleSubmit}
      />
      <DisplaytTasks tasks={tasks} />
      <CountTask count={count} />
    </>
  );
}

export default App;
