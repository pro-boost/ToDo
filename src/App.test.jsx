import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import App from "./App";

describe("App component", () => {
  // Renders the App component before each test
  beforeEach(() => {
    render(<App />);
  });

  it("should match the initial snapshot", () => {
    // Capture the initial snapshot of the rendered component
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it("should add a task and update the count on form submission", () => {
    // Find the input field and submit button
    const inputElement = screen.getByPlaceholderText(/Insert Task/);
    const submitButton = screen.getByText(/Add Task/);

    // Simulate user input and form submission
    fireEvent.change(inputElement, { target: { value: "New Task" } });
    fireEvent.click(submitButton);

    // Verify the task was added
    expect(screen.getByText("New Task")).toBeInTheDocument();

    // Verify the count was updated
    expect(screen.getByText("You have 1 task left")).toBeInTheDocument();
  });
  it("should not add a task if the input is empty", () => {
    const inputElement = screen.getByPlaceholderText(/Insert Task/);
    const submitButton = screen.getByText(/Add Task/);

    fireEvent.change(inputElement, { target: { value: "" } });
    fireEvent.click(submitButton);
    expect(screen.getByText("You have 0 tasks left")).toBeInTheDocument();
    expect(screen.queryByText("New Task")).not.toBeInTheDocument();
  });
});
