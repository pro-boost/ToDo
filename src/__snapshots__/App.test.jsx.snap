// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > should match the initial snapshot 1`] = `
<div>
  <h1>
    To Do List :
  </h1>
  <form
    role="form"
  >
    <input
      name="To Do"
      placeholder="Insert Task"
      type="text"
      value=""
    />
    <input
      type="submit"
      value="Add Task"
    />
  </form>
  <ul />
  <p>
    You have 
    0
     
    tasks
     left
  </p>
</div>
`;
